From 1aad063736f9ccaf7cdc7dab3db5ce5a371579b8 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 20 Aug 2018 14:08:36 -0700
Subject: [PATCH] TRITON-707 CNS crash on access to query.src because redis
 claim() took too long, remote closed TCP connection Reviewed by: Robert
 Mustacchi <rm@joyent.com>

---
 lib/dns-server.js | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/dns-server.js b/lib/dns-server.js
index c4c323b..ce099cf 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -236,6 +236,12 @@ DNSServer.prototype.handleQuery = function (q, cb) {
 
 	q.log.trace('begin query');
 
+	if (q.src === undefined || q.src === null) {
+		q.log.warn('dropping query from already-closed connection');
+		cb();
+		return;
+	}
+
 	var name = q.name().toLowerCase();
 	var type = q.type();
 	var addr = q.src.address;
-- 
2.21.0

